### PR TITLE
Describe results of OpenTelemetry PoC for tracing

### PR DESCRIPTION
**Pull request by chrkl**
**Description**

Add summary and examples of OpenTelemetry Tracing PoC

**Related issue(s)**
See also https://github.com/kyma-project/kyma/issues/10874, https://github.com/kyma-project/kyma/issues/10873

### CODE REVIEW

File: internal/research/opentelemetry/instrumentation-tracing.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+```
+var span trace.Span
+ctx := r.Context()
+ctx, span = h.tracer.Start(ctx, "myRequestHandler")
+defer span.End()
+...
+```
+
+Spans can be enhanced with additional events. Events can contain custom attributes. E.g.:
```
Comment (NHingerl) [inline internal/research/opentelemetry/instrumentation-tracing.md:73]:
```suggestion
Spans can be enhanced with additional events. Events can contain custom attributes, for example:
```

File: internal/research/opentelemetry/instrumentation-tracing.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+```
+...
+span.AddEvent("Acquiring lock", trace.WithAttributes(label.Int("pid", 1234))
+mutex.Lock()
+```
+
+## Automatic Instrumentation
+
+OpenTelemetry provides a list of [packages](https://github.com/open-telemetry/opentelemetry-go-contrib/tree/main/instrumentation) that can instrument specific Golang packages automatically. E.g., `net/http`.
```
Comment (NHingerl) [inline internal/research/opentelemetry/instrumentation-tracing.md:83]:
```suggestion
OpenTelemetry provides a list of [packages](https://github.com/open-telemetry/opentelemetry-go-contrib/tree/main/instrumentation) that can instrument specific Golang packages automatically, for example, `net/http`.
```

File: internal/research/opentelemetry/instrumentation-tracing.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+## Findings
+
+* The trace context propagation feature lacks documentation. We were not able to propagate the B3 context that has been injected by Istio to the OpenTelemetry SDK to the own span.
```
Comment (NHingerl) [inline internal/research/opentelemetry/instrumentation-tracing.md:87]: what is "the own span" ? Our span?

File: internal/research/opentelemetry/instrumentation-tracing.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+* Automatic instrumentation of the `net/http` packages did not generate any traces.
+* Documentation for the metrics API and logs API yet. In particular the enhancement of trace spans with metrics is not available.
```
Comment (NHingerl) [inline internal/research/opentelemetry/instrumentation-tracing.md:89]:
```suggestion
* Documentation for the metrics API and logs API yet. In particular, the enhancement of trace spans with metrics is not available.
```
I don't understand the first sentence, some information seems to be missing.

File: internal/research/opentelemetry/instrumentation-tracing.md [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+* Traces were exported to the collector unreliably. Needs further investigation
```
Comment (NHingerl) [inline internal/research/opentelemetry/instrumentation-tracing.md:90]:
```suggestion
* Traces were exported to the collector unreliably. Needs further investigation.
```

**Review by NHingerl — CHANGES_REQUESTED** [review]

File: internal/research/opentelemetry/instrumentation-tracing.md (hunk 3 above)
Comment (chrkl) [reply internal/research/opentelemetry/instrumentation-tracing.md]: Then span that has been created for the current request by the SDK. Changed the wording a bit to make this clear.

File: internal/research/opentelemetry/deployment-tracing.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+* Kafka
+* OpenCensus
+* OTLP (grpc and http)
+* Zipkin
+
+Traces can be exported using the same APIs as well as writing to STDOUT and files.
+
+### Collector Deployment
+
+The collector can be deployed as a central service, a DaemonSet or sidecar container to each pod. Telemetry data can be forwarded by an exporter to another collector or an external service (e.g., Jaeger).
```
[the remark this answers is not in the document]
Comment (chrkl) [reply internal/research/opentelemetry/deployment-tracing.md]: One of the options has to be used. Rephrased the sentences a bit.

**Review by NHingerl — APPROVED** [review]

**Review by alexandra-simeonova — APPROVED** [review]

**Pull request merged**